uptane-standard.md

```diff
@@ -780,15 +787,19 @@ The ECU SHALL verify that the latest image matches the latest metadata as follow
 
 If the ECU has secondary storage, the checks SHOULD be performed on the image in secondary storage, before it is installed.
 
-If any step fails, the ECU SHALL jump to the fifth step ({{create_version_report}}). If a step fails and the ECU does not have secondary storage, and the ECU has created a backup of its previous working image, the ECU SHOULD now install the backup image.
+If any step fails, the ECU SHALL jump to the ({{create_version_report}}) step. If an ECU does not have secondary storage, then before attempting verification, the ECU SHALL ensure a backup of a previous working image and associated metadata is created. This MAY be stored on the updating ECU or elsewhere (e.g., associated primary). Thus, if a step fails and the ECU does not have secondary storage, the ECU SHALL install the backup image to ensure it is in a working condition.
```

Comment:
Sorry, but this doesn't make sense. If the ECU does _not_ have secondary storage, how SHALL it ensure a backup of the previous working image on itself, if not elsewhere?

Reply:
Also, which step should fail before the ECU installs the backup image? What if the new image hasn't been replaced yet?

Reply:
I have concerns about the ECU's ability to ensure a backup is there and valid if it is not doing it itself. How would it ensure the integrity of the image or that it was stored properly?

Reply:
Thanks for the comments, Phil.

I agree this is a difficult problem to solve.
Overall, I think everyone can agree that the vehicle needs to be in a 'functioning state' in case of update failure.
I am not saying the approach I have outlined above is optimal, but at least it starts a conversation on how to ensure a fail-back mode is possible for partial-verification (pv) secondaries which do not contain additional storage for previous updates (i.e., A/B partition not available).

My approach was to ensure a couple of important things happened: 
    1. Backup the currently working image & metadata before attempting update
    [Failure occurs during update process]
    2. Update pv-secondary to the previously working version 
    3. Report this failure to the server (not currently in PR, but probably should be expanded to include)

To answer your question, Phil, maybe we can be more verbose in a couple of areas.
For example, in Step 1^, document that the storage location (e.g., Primary), has notified the pv-secondary that storage was successfully completed, before pv-secondary begins update process.
For ensuring the integrity, maybe the approach we take is requiring pv-secondaries to store a hash of its previously working image, which is not overwritten until a successful update has occurred. 
The pv-secondary can use this stored hash to verify against hash in metadata & image received from associated update provider (e.g., primary).

These ^^ don't have to be the concrete solution or wording we put into the standard, but I do hope this facilitates additional feedback which we can include in v1.0 to ensure the standard accounts for failure cases.